File: python/baiduads-sdk/baiduads_manual/videoupload/api/video_upload_service.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Copyright (c) 2022 Baidu.com, Inc. All Rights Reserved
"""

from baiduads_manual.upload.upload_client import UploadClient
from baiduads_manual.videoupload.model.add_video_response_wrapper import AddVideoResponseWrapper


class VideoUploadService:
    """VideoUploadService
        please visit:  https://dev2.baidu.com
    """

    def __init__(self):
        self.upload_client = UploadClient()

    def add_video(self, add_video_request_wrapper, **kwargs):
        """

        Args:
            add_video_request_wrapper:AddVideoRequestWrapper
            **kwargs:

        Returns:
            AddVideoResponseWrapper
        """
        path = "/json/sms/service/VideoUploadService/addVideo"
        header = add_video_request_wrapper.header
        request_body = add_video_request_wrapper.body
        file = request_body.file
        signature = request_body.signature
        params = request_body.params
        timeout = request_body.timeout

        # form params
        form = [("signature", signature)]
        # set header param
        attribute_map = header.attribute_map
        data_dict = header.to_dict()
        for k, v in data_dict.items():
            form.append((attribute_map.get(k, k), v))

        upload_options = request_body.uploadOptions
        multipartThreshold = None
        partSize = None
        partParallel = None
        if upload_options is not None:
            multipartThreshold = upload_options.multipartThreshold
            partSize = upload_options.partSize
            partParallel = upload_options.partParallel

        return self.upload_client.upload_file(
            path,
            file,
            form,
            params.__dict__,
            multipartThreshold=multipartThreshold,
            partSize=partSize,
            partParallel=partParallel,
            timeout=timeout,
            returnType=AddVideoResponseWrapper
        )

File: java/baiduads-sdk/src/main/java/com/baidu/dev2/api/sdk/manual/videoupload/api/VideoUploadService.java
package com.baidu.dev2.api.sdk.manual.videoupload.api;

import com.baidu.dev2.api.sdk.common.ApiRequestHeader;
import com.baidu.dev2.api.sdk.invoke.ApiException;
import com.baidu.dev2.api.sdk.manual.upload.UploadClient;
import com.baidu.dev2.api.sdk.manual.upload.model.MultiUploadOptions;
import com.baidu.dev2.api.sdk.manual.videoupload.model.AddVideoParams;
import com.baidu.dev2.api.sdk.manual.videoupload.model.AddVideoRequest;
import com.baidu.dev2.api.sdk.manual.videoupload.model.AddVideoRequestWrapper;
import com.baidu.dev2.api.sdk.manual.videoupload.model.AddVideoResponseWrapper;
import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.node.ObjectNode;

import java.lang.reflect.Field;
import java.lang.reflect.Modifier;
import java.util.HashMap;
import java.util.Map;

public class VideoUploadService {


    public VideoUploadService() {

    }


    public AddVideoResponseWrapper addVideo(AddVideoRequestWrapper addVideoRequestWrapper) throws ApiException {
        if (addVideoRequestWrapper == null) {
            throw new ApiException(400,
                    "Missing the required parameter 'addVideoRequestWrapper' when calling addVideo");
        }

        // 文件限制 100MB
        if (addVideoRequestWrapper.getBody().getFile().length() > 100 * 1024 * 1024) {
            throw new ApiException("file size too big");
        }

        String localVarPath = "/json/sms/service/VideoUploadService/addVideo";

        Map<String, Object> formParams = new HashMap<String, Object>();
        ObjectNode objectNode = null;
        Integer partParallel = null;
        Integer partSize = null;
        Integer multipartThreshold = null;
        try {
            // set header params
            ApiRequestHeader header = addVideoRequestWrapper.getHeader();
            Field[] declaredFields = ApiRequestHeader.class.getDeclaredFields();
            for (Field field : declaredFields) {
                if ((Modifier.STATIC & field.getModifiers()) != Modifier.STATIC) {
                    field.setAccessible(true);
                    String name = field.getName();
                    Object value = field.get(header);
                    formParams.put(name, value);
                }
            }

            // set body params
            AddVideoRequest body = addVideoRequestWrapper.getBody();
            String signature = body.getSignature();
            formParams.put("signature", signature);
            AddVideoParams params = body.getParams();
            ObjectMapper objectMapper = UploadClient.INSTANCE.getApiClient().getObjectMapper();
            objectNode = objectMapper.convertValue(params, ObjectNode.class);
            MultiUploadOptions uploadOptions = body.getUploadOptions();
            if (uploadOptions != null) {
                partParallel = uploadOptions.getPartParallel();
                partSize = uploadOptions.getPartSize();
                multipartThreshold = uploadOptions.getMultipartThreshold();
            }
        } catch (Exception e) {
            throw new ApiException(e);
        }


        return UploadClient.INSTANCE.uploadFile(
                localVarPath,
                formParams,
                addVideoRequestWrapper.getBody().getFile(),
                objectNode,
                null,
                multipartThreshold,
                partSize,
                partParallel,
                addVideoRequestWrapper.getBody().getTimeout(),
                new TypeReference<AddVideoResponseWrapper>() {
                }
        );
    }

}
